src/model/version_list.rs

```diff
@@ -3,10 +3,35 @@ pub struct VersionList {}
 
 impl VersionList {
     pub fn latest_version(versions: Vec<Version>) -> Option<Version> {
-        let mut sorted_versions = versions.to_vec();
-        sorted_versions.sort_by(|a, b| b.uploaded_at.cmp(&a.uploaded_at));
+        let sorted_versions = VersionList::sort_versions(versions);
         sorted_versions.first().map(|v| v.clone())
     }
+
+    pub fn latest_version_of_distribution_group(versions: Vec<Version>, distribution_group: String) -> Option<Version> {
+        let sorted_versions = VersionList::sort_versions(versions);
+
+        for version in sorted_versions.iter() {
+            match &version.distribution_groups {
+                Some(groups) => {
+                    for group in groups.iter() {
```

Comment:
Can you rewrite the internal loop using filter? Should be something like
```
groups.iter().filter(|group|, group.name == distribution_group)
```

Reply:
I essence the version list is [filtered](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter) for a matching version, I'd try to filter `sorted_versions` already, aka begin with `sorted_versions.iter().filter...`

Reply:
Yes, I thought about that @q231950. However, distribution groups are optional for versions, which is why I think you need the match. Am I wrong?
```
pub distribution_groups: Option<Vec<DistributionGroup>>
```

Reply:
I just tried and made it work using filter, but it was admittedly ugly.

I had some thoughts about how `version` and `distribution_group` are both filters which could be applied to the version list. That could make parts of the code more readable, but I'm happy with any of these options as they are now.

One thing I noticed was that tests are not even compiling any more. I guess that and some additional tests for the distribution group as @jamuc had mentioned would be great 👌

Reply:
Some thought food for you @huNt-FMJ - I'm sure it could be improved, but anyways:
```rust
pub fn latest_version_of_distribution_group(
        versions: Vec<Version>,
        distribution_group: String,
    ) -> Option<Version> {
        let sorted_versions = VersionList::sort_versions(versions);

        let group_version = sorted_versions
            .iter()
            .filter(|version| VersionList::by_distribution_group(&distribution_group, version.distribution_groups.as_ref()))
            .next();

        return group_version.cloned();
    }

    fn by_distribution_group(distribution_group: &String, distribution_groups: Option<&Vec<DistributionGroup>>) -> bool {
        match distribution_groups {
            Some(groups) => groups.iter().any(|group| group.name == distribution_group.to_string()),
            None => false,
        }
    }
```